beacon-chain/state/getters.go

```diff
@@ -498,3 +497,36 @@ func CopyETH1Data(data *ethpb.Eth1Data) *ethpb.Eth1Data {
 
 	return newETH1
 }
+
+// CopyPendingAttestation copies the provided pending attestation object.
+func CopyPendingAttestation(att *pbp2p.PendingAttestation) *pbp2p.PendingAttestation {
+	aggBytes := []byte(att.AggregationBits)
+	newBitlist := make([]byte, len(aggBytes))
+	copy(newBitlist, aggBytes)
+	blockRoot := [32]byte{}
+	copy(blockRoot[:], att.Data.BeaconBlockRoot)
+	data := &ethpb.AttestationData{
+		Slot:            att.Data.Slot,
+		CommitteeIndex:  att.Data.CommitteeIndex,
+		BeaconBlockRoot: blockRoot[:],
+		Source:          CopyCheckpoint(att.Data.Source),
+		Target:          CopyCheckpoint(att.Data.Target),
+	}
+	return &pbp2p.PendingAttestation{
+		AggregationBits: newBitlist,
+		Data:            data,
+		InclusionDelay:  att.InclusionDelay,
+		ProposerIndex:   att.ProposerIndex,
+	}
+}
+
+// CopyCheckpoint copies the provided checkpoint.
+func CopyCheckpoint(cp *ethpb.Checkpoint) *ethpb.Checkpoint {
```

Comment:
Can you use this for the other check points too?

Reply:
done